hi-ml/src/health_ml/eval_runner.py

```diff
@@ -0,0 +1,29 @@
+from pytorch_lightning import LightningDataModule
+
+from health_azure.logging import logging_section
+from health_ml.runner_base import RunnerBase
+
+
+class EvalRunner(RunnerBase):
+    """A class to run the evaluation of a model on a new dataset. The initialization logic is taken from the base
+    class `RunnerBase`.
+    """
+
+    def validate(self) -> None:
+        """Checks if the fields of the class are set up correctly."""
+        if self.container.src_checkpoint is None or self.container.src_checkpoint.checkpoint == "":
+            raise ValueError(
+                "To use model evaluation, you need to provide a checkpoint to use, via the --src_checkpoint argument."
+            )
+
+    def run(self) -> None:
+        """Start the core workflow that the class implements: Initialize the trainer and run inference on the
```

Comment:
To the untrained eye it might be confusing why we'd need to 'initialize the trainer' here in the EvalRunner - can we explain why this is necessary here? Also we could mention that if the container has an extra_val_epoch method it will be run first before inference starts

Reply:
Oh I don't thing extra_val_epoch is run here, only run_infererence, this is for an evaluation on a external dataset so all data goes into the test_datamodule

Reply:
But `init_inference` is called here and within that (lines 204-205 of runner_base.py), `on_run_extra_validation_epoch` is called (if `run_extra_val_epoch`)...or am I missing something?

Reply:
correct, but this is only to tell the container to switch some plot options, the extra val epoch is run by calling run_validation

Reply:
I see!

Reply:
Good point, @mebristo . I've cleaned that up, now there is no such hook run when doing eval on a new dataset.